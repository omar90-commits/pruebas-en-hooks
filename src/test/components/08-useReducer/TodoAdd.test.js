import React from 'react';
import { TodoAdd } from '../../../components/08-useReducer/TodoAdd';
import { shallow } from 'enzyme';

describe('Pruebas en <TodoAdd />', () => {
	
	const handleAddTodo = jest.fn();

	const wrapper = shallow( <TodoAdd handleAddTodo={handleAddTodo} /> );

	test('Debe de mostrarse correctamente', () => {
		
		expect( wrapper ).toMatchSnapshot();
	});

	test('No debe de llamar la funcion handleAddTodo', () => {
		
		wrapper.find('form').simulate('submit', { preventDefault(){} });

		expect( handleAddTodo ).toHaveBeenCalledTimes(0);
	});

	test('Debe de llamar la funcion handleAddTodo', () => {
		
		const value = 'Hola Mundo';
		const input = wrapper.find('input');

		input.simulate('change', { target: { value, name:'description' } });
		wrapper.find('form').simulate('submit', { preventDefault(){} });

		expect( handleAddTodo ).toHaveBeenCalledTimes(1);
		expect( handleAddTodo ).toHaveBeenCalledWith( expect.any(Object) );
		expect( handleAddTodo ).toHaveBeenCalledWith( {
			desc: value,
			done: false,
			id: expect.any(Number),
		});

		expect( wrapper.find('input').prop('value') ).toBe('');
	});
});