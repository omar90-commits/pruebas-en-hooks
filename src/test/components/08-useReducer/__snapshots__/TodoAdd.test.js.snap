// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender ..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;

exports[`Pruebas en <TodoApp /> Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    TodoApp ( 
    0
     ) 
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <TodoList
        handleDelete={[Function]}
        handleToggle={[Function]}
        todos={Array []}
      />
    </div>
    <div
      className="col-5"
    >
      <TodoAdd
        handleAddTodo={[Function]}
      />
    </div>
  </div>
</div>
`;
